pkgs/tools/networking/boundary/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, lib, fetchurl, unzip }:
+let
```

Comment:
Can you resolve this let in? It makes the package way more complicated than it needs to be.

Reply:
What do you mean by "resolve this let in"?
This is how plenty of pre-built packages fetch dependencies.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/virtualization/firecracker/default.nix#L3-L27
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/vscodium.nix#L3-L28
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/p3x-onenote/default.nix#L3-L31
https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/broadcom-sta/default.nix#L3-L13

If there's a better way I'm all ears

Reply:
> What do you mean by "resolve this let in"?

Replace it with rec and inline the lines.

I would prefer it the version and fetchsrc would be in the mkderivation like normal packages.

Reply:
since fetchsrc is a function it wouldn't let me just do

```
fetchsrc = sha256: fetchurl {
      url = "https://releases.hashicorp.com/boundary/${version}/boundary_${version}_${suffix}.zip";
      sha256 = sha256."${system}";
    };
```

inside the `mkDerivation`

```
error: --- TypeError ----------------------------------------------------------------------------------------------- nix-build
at: (192:11) in file: /home/user/projects/personal/nixpkgs/pkgs/stdenv/generic/make-derivation.nix

   191|         // (lib.optionalAttrs (!(attrs ? name) && attrs ? pname && attrs ? version)) {
   192|           name = "${attrs.pname}-${attrs.version}";
      |           ^
   193|         } // (lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform && !dontAddHostSuffix && (attrs ? name || (attrs ? pname && attrs ? version)))) {

cannot coerce a function to a string
(use '--show-trace' to show detailed location information)
```

I've added 3 temporary commits with different setups
One with fetchsrc still outside and 2 with it in the `let` of `src`
Let me know which is closest to what you're looking for

[`5a0375b` (#111151)](https://github.com/NixOS/nixpkgs/pull/111151/commits/5a0375bb66ad4ebfcd99f775df7e5071770a99a6)
[`f997c92` (#111151)](https://github.com/NixOS/nixpkgs/pull/111151/commits/f997c9210c04df2cb918d6ca57cc86b9d06eb93a)
[`0c7e521` (#111151)](https://github.com/NixOS/nixpkgs/pull/111151/commits/0c7e52171166177b1a2d15392285106a33052a97)

Reply:
I think [`5a0375b` (#111151)](https://github.com/NixOS/nixpkgs/pull/111151/commits/5a0375bb66ad4ebfcd99f775df7e5071770a99a6) is the cleanest way but I thought it could be resolved more cleanly. Do what you want. The difference is marginally-